packages/x-data-grid/src/components/grid/toolbar/GridToolbarRoot.tsx

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable react/no-unused-prop-types */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { styled, SxProps, Theme } from '@mui/material/styles';
+import composeClasses from '@mui/utils/composeClasses';
+import { getDataGridUtilityClass } from '../../../constants/gridClasses';
+import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
+import {
+  useGridComponentRenderer,
+  RenderProp,
+} from '../../../hooks/utils/useGridComponentRenderer';
+
+export interface GridToolbarRootProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The system prop that allows defining system overrides as well as additional CSS styles.
+   */
+  sx?: SxProps<Theme>;
+  /**
+   * A function to customize rendering of the component.
+   */
+  render?: RenderProp<{}>;
+}
+
+type OwnerState = DataGridProcessedProps;
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['toolbarRoot'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const StyledGridToolbarRoot = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'ToolbarRoot',
+})<{ ownerState: OwnerState }>(({ theme }) => ({
```

Comment:
Would it make sense to consider the toolbar as a new component, `name = MuiDataGridToolbar, slot = Root`? The term "root" is usually used only for root elements in material-ui.